app/bundles/CoreBundle/Controller/RobotsTxtController.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * @copyright   2016 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\CoreBundle\Controller;
+
+use Mautic\CoreBundle\CoreEvents;
+use Mautic\CoreBundle\Event\BuildRobotsTxtEvent;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Class RobotsTxtController.
+ */
+class RobotsTxtController extends CommonController
+{
+    /**
+     * @return Response
+     */
+    public function indexAction()
+    {
+        $dispatcher = $this->dispatcher;
+        $debug      = $this->factory->getKernel()->isDebug();
+        $event      = new BuildRobotsTxtEvent($this->getDefaultContent(), $debug);
```

Comment:
Can you move this logic out of the Controller please? I know that Controllers in Mautic holds lot of logic, but we want to keep it clean for new ones. It will help us for Mautic 3.

Just create a new class, inject Dispatcher and move all logic there. Controller will only call this class for a Response.

Thank you

Reply:
Also - Unit test will be easy to write after this change.

Reply:
Can you show me example of that class? It would help me :)

Reply:
Take a look at this Controller:
https://github.com/mautic/mautic/blob/staging/app/bundles/ReportBundle/Controller/ScheduleController.php

Just create a new class and move this code there:
```
$event = new BuildRobotsTxtEvent($this->getDefaultContent(), $debug);
$this->dispatcher->dispatch(CoreEvents::BUILD_MAUTIC_ROBOTS_TXT, $event);

return $event->getContent();
```
```
private function getDefaultContent() {
...
```

Note that this line is redundant and can be removed:
`if ($this->dispatcher->hasListeners(CoreEvents::BUILD_MAUTIC_ROBOTS_TXT)) {`

Also, function `getDefaultContent` should be a separate class in ideal code, but I think there is no real need of it because class is very simple and content is hardcoded here now. It will be easy enough to write unit test.

`RobotsTxtController` will only have a few lines of code after this change:
```
$content = $robotTxtService->getContent();

return new Response($content);
```

Ping me if anything is not clear.